.github/workflows/test.yml

```diff
@@ -1,54 +1,133 @@
+        run: composer --working-dir=html require ${{ github.repository }}:"dev-${GIT_BRANCH} as 1.0.x-dev"
+
+  phpcs:
+    name: Coding standards checks
+    needs: build
+    runs-on: ubuntu-latest
+      
+    steps:
+
+      - name: Cached workspace
+        uses: actions/cache@v2
+        with:
+          path: ./html
+          key: ${{ runner.os }}-localgov-build-${{ github.run_id }}
+          restore-keys: |
+            ${{ runner.os }}-localgov-build-
+      
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.4'
+
+      - name: Run coding standards checks
         run: |
-          echo "GIT_BRANCH=${GITHUB_HEAD_REF}" >> $GITHUB_ENV
-          echo "GIT_REPO=${{ github.event.pull_request.head.repo.full_name }}" >> $GITHUB_ENV
+          cd html
+          ./bin/phpcs -p
+
+  phpstan:
+    name: Deprecated code checks
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Cached workspace
+        uses: actions/cache@v2
+        with:
+          path: ./html
+          key: ${{ runner.os }}-localgov-build-${{ github.run_id }}
+          restore-keys: |
+            ${{ runner.os }}-localgov-build-
+      
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.4'
 
-      - name: Setup package source and authentication for the test target
+      - name: Run deprecated code checks
         run: |
-          composer --working-dir=html config repositories.1 vcs git@github.com:${GIT_REPO}.git
-          composer global config github-oauth.github.com ${{ github.token }}
+          cd html
+          ./bin/phpstan analyse -c ./phpstan.neon ./web/profiles/contrib/localgov/ ./web/modules/contrib/localgov_*
 
-      - name: Grab test target from this repo
-        run: composer --working-dir=html require localgovdrupal/${{ github.event.repository.name }}:"dev-${GIT_BRANCH} as 1.0.x-dev"
+  phpunit:
+    name: PHPUnit tests
+    needs: build
+    runs-on: ubuntu-latest
+    
+    steps:
+
+      - name: Clone drupal_container
+        uses: actions/checkout@v2
+        with:
+          repository: localgovdrupal/drupal-container
+          ref: master
+
+      - name: Cached workspace
+        uses: actions/cache@v2
+        with:
+          path: ./html
+          key: ${{ runner.os }}-localgov-build-${{ github.run_id }}
+          restore-keys: |
+            ${{ runner.os }}-localgov-build-
 
       - name: Start Docker environment
-        run: docker-compose up -d
+        run: docker-compose -f docker-compose.yml up -d
 
-      - name: Run tests
-        run: ./run-tests.sh
-        shell: bash
+      - name: Run PHPUnit tests
+        run: |
+          mkdir -p ./html/web/sites/simpletest && chmod 777 ./html/web/sites/simpletest
+          sed -i "s#http://localgov.lndo.site#http://drupal#; s#mysql://database:database@database/database#sqlite://localhost/dev/shm/test.sqlite#" ./html/phpunit.xml.dist
```

Comment:
A crucial slash may have disappeared from ```localhost//dev/shm/test.sqlite```.  Please compare this bash snippet with [run-test.sh](https://github.com/localgovdrupal/drupal-container/blob/master/run-tests.sh)

Reply:
The extra slash is not actually needed. I discovered this while reading the Drupal docs on this: https://www.drupal.org/docs/automated-testing/phpunit-in-drupal/phpunit-browser-test-tutorial#s-sqlite

Reply:
Hi Steve, you correct that you can ignore the extra slash and the test will run just fine.  What happens with "sqlite://localhost/dev/shm/test.sqlite" is, a file called "dev/shm/test.sqlite" is created **inside** Drupal's docroot.  But if you use "sqlite://localhost//dev/shm/test.sqlite", you genuinely get /dev/shm/test.sqlite which speeds up the test run slightly.

I can see that even with "sqlite://localhost/dev/shm/test.sqlite", the PHPUnit tests have taken 16m47s which is not bad. This used to take about 14m in [previous test runs](https://github.com/localgovdrupal/localgov_alert_banner/runs/2468599165).  Not a lot of difference.  So I will leave this to you.

Reply:
@Adnan-cds  I've added the extra slash and the run took 14m19s. Nice catch.

Are you happy to approve this now?